### ISSUE TITLE
Autocomplete battletags

### ISSUE THREAD

**Issue by cheshire137** [cheshire137/overwatch-team-comps#100]
@RobThePM and @justbarreto, I want your opinions on an idea I just had. We've already discussed letting team organizers see the hero pools of their team by way of ordering the hero select based on those hero pools, if I specify the battletag for my player and that user happens to have signed in and set their hero pools. So Zion signs in and sets Genji to Grandmaster, Rob adds a player with Zion's battletag to his team comp, then Rob sees Genji at the top of the list for Zion when choosing a hero for him to play.

However, I think the part where the team organizer specifies battletags for his players might be burdensome. I personally don't know all my friends' battletags; I know the first part, but I don't know the digits at the end. What do you think about when a team organizer goes to specify battletags for her players, we offer autocompletion as the user types based on the battletags already in the system? So Rob puts "Sarah" on his team, then he goes to fill out her battletag and as he types "ch", he sees "cheshire137#1695" show up as an option.

I don't think this would be hard to implement, it's just a matter of do we want to expose battletags of potentially unrelated users to team organizers. So Rob could start typing "ch" and see "cheetos#1234", some rando he's never heard of. I guess all that Rob learns from this is some Battle.net user with that battletag happens to use the app, and if Rob puts that player in his team comp, he could see cheetos' hero pool.

**Comment by cheshire137** [comment]
@rewinfrey you might also have thoughts about this!

We could make it an opt-in thing, like on user signup we ask the user "do you want other users to be able to search by your battletag and find you." We could also just add a privacy policy saying "hey if you sign up, randos can potentially see your battletag and by extension your hero pool."

**Comment by RobThePM** [comment]
I agree, it can be burdensome to add your teammate'a battle tag manually. Great job thinking about this.

My initial thoughts...

My experience with auto complete fields is that when the list gets long, it has a negative impact on peformance (e.g. the app or typing will freeze as it tries to look up). If we do this, will it have a negative impact on performance speed?

A workaround for a user could simply be to not enter a real battle tag and just enter instead a placeholder, at least temporarily. For example, if I didn't know Zion's battle tag, I could just type in "Zion" in the player name field. I wouldn't see his preferences, but I can still set the heroes. I would be okay with this or what you proposed above.

What if instead of exposing the entire database to a user, we only show them the battle tags of players on their friends list? Is that possible?

**Comment by RobThePM** [comment]
I'm also okay with waiting and solving for this in Version 2.

**Comment by cheshire137** [comment]
> What if instead of exposing the entire database to a user, we only show them the battle tags of players on their friends list?

Oh that sounds much better! If the friends list is available to us via the Battle.net API, we could even suggest players as the team organizer types player names. Like I'm brand new to the app, I start typing "zi" in the player name field, and it prompts me with "ZionCross#1234" from my friends list. If I choose that suggestion, it creates a player with that battletag and "ZionCross" as the name.

**Comment by cheshire137** [comment]
Blizzarddddddddd... 

> At the moment its not possible to retrieve an account's friends list via the public APIs. 

-- https://us.battle.net/forums/en/bnet/topic/14607771239#post-4

**Comment by RobThePM** [comment]
Awe, oh well. I think it's okay to hold off on this until a later version. However, you're ultimately doing the work, so totally up to you if you want to do this in Version 1 or not.

**Comment by cheshire137** [comment]
> If we do this, will it have a negative impact on performance speed?

I don't think it would. We wouldn't make a request to the back end every time the user types a character; we'd wait a half second after the user finishes typing to submit the request, so things would stay responsive and we wouldn't be sending a bajillion requests to the back end.
